api/v1alpha4/virtualmachine_types.go

```diff
@@ -674,9 +674,48 @@ type VirtualMachineSpec struct {
 	// +listType=map
 	// +listMapKey=name
 
-	// Volumes describes a list of volumes that can be mounted to the VM.
+	// Volumes describes the desired list of volumes to attach to the VM.
 	Volumes []VirtualMachineVolume `json:"volumes,omitempty"`
 
+	// +optional
+	// +listType=map
+	// +listMapKey=busNumber
+	// +kubebuilder:validation:MaxItems=2
+
+	// IDEControllers describes the desired list of IDE controllers for the VM.
```

Comment:
We talked about this in person but leaving a comment for others to chime in - maybe move these under a new `Spec.Hardware` would be good for if we have to later add more controller types or other HW'y things (like possibility moving `Spec.Cdrom` under this new container too)

Reply:
Great point. Done and one.